### DISCUSSION TITLE
Samba Share

### DISCUSSION THREAD

**Discussion by doitandbedone** [doitandbedone/ispyagentdvr-docker#175] [General]
Originally posted by @HSB-collab:

Hi thanks for the Docker Container. Like @giube above, I would like to be able to write my camera recordings to a samba network share. I have set up the samba share using the docker command with the -v option to map my network share to the docker directory location for the media as follows:
docker -v //LAN_network_address_of_share/media:/agent/Media/WebServerRoot/Media/
When I list the volumes in the docker container (using docker inspect -f '{{ .Mounts }}' containerid), I get the following:
{bind /LAN_network_address_of_share/media /agent/Media/WebServerRoot/Media true rprivate}
Note that there is only one forward slash in front of the network address in the bind. I am not sure if this is significant? I also don't know the meaning of 'rprivate'.
However, when I record a video in Agent DVR, it only records to the directory in the Docker container. I have checked that the network share is mounted and that it is writable from the host machine on which the Agent DVR is running (I can write files using File Manager on the docker host), and the permissions are correct.
I have spent hours trying to figure this out but with no success so any suggestions would be greatly appreciated!

**Comment by doitandbedone** [comment]
Hello @HSB-collab, according to docker documentation, rprivate us part of [bind-propagation](https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation) and is listed as:

> The default. The same as private, meaning that no mount points anywhere within the original or replica mount points propagate in either direction.

You can try this: 
https://www.digitalocean.com/community/tutorials/how-to-share-data-between-docker-containers
Also a bit on the different types of volumes:
https://spin.atomicobject.com/2019/07/11/docker-volumes-explained/
I've only did it once but was for testing purposes though. But usually you use named volumes. Check it out and hope it helps!

You can find more about docker volumes here:
https://docs.docker.com/storage/bind-mounts/

**Reply by HSB-collab** [reply]
Thanks. I am afraid I gave up on the docker containers and just installed natively on Ubuntu which now works.

**Comment by doitandbedone** [comment]
They can be tricky to work with at times. At least you have it working
natively.
